Spec/RestClient.swift

```diff
@@ -485,6 +485,63 @@ class RestClient: QuickSpec {
                     expect(NSRegularExpression.match(testHTTPExecutor.requests[1].URL!.absoluteString, pattern: "//[a-e].ably-rest.com")).to(beTrue())
                 }
 
+                // RSC15d
+                context("should use an alternative host when") {
+                    for caseTest: NetworkAnswer in [.HostUnreachable,
+                                                    .RequestTimeout(timeout: 0.1),
+                                                    .HostInternalError(code: 501)] {
+                        pending("\(caseTest)") {
+                            let options = ARTClientOptions(key: "xxxx:xxxx")
+                            let client = ARTRest(options: options)
+                            client.httpExecutor = testHTTPExecutor
+                            testHTTPExecutor.http = MockHTTP(network: caseTest)
+                            let channel = client.channels.get("test")
+
+                            testHTTPExecutor.afterRequest = { _ in
+                                if testHTTPExecutor.requests.count == 2 {
+                                    // Stop
+                                    testHTTPExecutor.http = nil
+                                }
+                            }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                channel.publish(nil, data: "nil") { _ in
+                                    done()
+                                }
+                            }
+
+                            expect(testHTTPExecutor.requests).to(haveCount(2))
+                            expect(NSRegularExpression.match(testHTTPExecutor.requests[0].URL!.absoluteString, pattern: "//rest.ably.io")).to(beTrue())
+                            expect(NSRegularExpression.match(testHTTPExecutor.requests[1].URL!.absoluteString, pattern: "//[a-e].ably-rest.com")).to(beTrue())
+                        }
+                    }
+                }
+
+                // RSC15d
+                it("should not use an alternative host when the client receives a bad request") {
+                    let options = ARTClientOptions(key: "xxxx:xxxx")
+                    let client = ARTRest(options: options)
+                    client.httpExecutor = testHTTPExecutor
+                    testHTTPExecutor.http = MockHTTP(network: .HostBadRequest)
```

Comment:
Ignore me, I see below... that is fine. Perhaps update the name to HostBadRequest400 to make it clearer for dummies like me :)